niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/SignaturePermissionTester.java

```diff
@@ -2550,35 +2656,60 @@ Transacts.cancelAuthenticationFromService, getActivityToken(),
-                    handlerThread.start();
-                    Handler handler = new Handler(handlerThread.getLooper());
-                    // An Exception array is used since local variables referenced from a lambda
-                    // expression must be effectively final.
-                    Exception[] caughtException = new Exception[1];
-                    CountDownLatch latch = new CountDownLatch(1);
-                    handler.post(() -> {
-                        PhoneStateListener listener = new PhoneStateListener() {
-                            @Override
-                            public void onSignalStrengthsChanged(SignalStrength signalStrength) {
-                                mLogger.logDebug("onSignalStrengthChanged: signalStrength = "
-                                        + signalStrength);
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                        //The way to using the listner was obsolated after android T, so let me choose using this option instead
+                        //https://source.corp.google.com/android-internal/cts/tests/tests/telephony/current/src/android/telephony/cts/TelephonyManagerTest.java;rcl=91e7eeb88fd7c7b2855068487f89af167cc79808;l=4186
+                        //problem : if the signature does not match it always fail,because the method also checks MODIFY_PHONE_STATE permission
+
+                        SignalStrengthUpdateRequest.Builder builder = new SignalStrengthUpdateRequest.Builder()
+                                .setSignalThresholdInfos(Collections.EMPTY_LIST);
+
+                        builder = (SignalStrengthUpdateRequest.Builder)
+                                invokeReflectionCall(SignalStrengthUpdateRequest.Builder.class,
+                                "setSystemThresholdReportingRequestedWhileIdle", builder,
+                                new Class<?>[]{boolean.class},true);
+                        SignalStrengthUpdateRequest request = builder.build();
+                        try {
+                            mTelephonyManager.setSignalStrengthUpdateRequest(request);
+                        } catch (IllegalStateException ex){
+                            mLogger.logInfo("Expected:"+ex.getMessage());
+                        }
+
+                    } else {
+                        HandlerThread handlerThread = new HandlerThread(TAG);
+                        handlerThread.start();
+                        Handler handler = new Handler(handlerThread.getLooper());
+                        // An Exception array is used since local variables referenced from a lambda
+                        // expression must be effectively final.
+                        //https://source.corp.google.com/tm-dev/cts/tests/tests/telephony/current/src/android/telephony/cts/PhoneStateListenerTest.java;l=333?q=LISTEN_ALWAYS_REPORTED_SIGNAL_STRENGTH&sq=package:tm-dev
```

Comment:
This should reference the publicly available source; should this be https://cs.android.com/android/platform/superproject/+/master:cts/tests/tests/telephony/current/src/android/telephony/cts/PhoneStateListenerTest.java;l=328;drc=e64188140ba71c7b7424b044119b37af1dde6609?

Reply:
Thanks. Fixed

Reply:
Did the commit make it into this pull request? I'm still seeing the previous link in my changes view.

Reply:
sorry. will update it.

Reply:
Update looks good to me, thanks!